File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/eth/block/EthGetStorageAtApiTest.java
package opensdk.sdk.apis.eth.block;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.EthGetStorageAtResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Eth RPC Test")
public class EthGetStorageAtApiTest {

    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC eth_getStorageAt")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        EthGetStorageAtResponse br = sdk.eth.getStorageAt(
            "0x295a70b2de5e3953354a6a8344e616ed314d7251",
            "0x0",
            "latest")
        .send();
        br.getResult();
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/eth/block/EthGetStorageAtApi.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('eth_getStorageAt API', () => {
    test('should return eth_getStorageAt', (done) => {

        let callbackOne = function (error, data, response) {

            expect(error).toBeNull();
            expect(data).toBeDefined()
            expect(data).toMatch(/^0x.*$/gm)
            done();
        };
        const address = '0x295a70b2de5e3953354a6a8344e616ed314d7251'
        const quantity = '0x0'
        const blockNumberOrHashOrTag = 'latest'
        sdk.eth.getStorageAt(address, quantity, blockNumberOrHashOrTag, {}, callbackOne);
    });
});
